sys_util/src/struct_util.rs

```diff
@@ -108,8 +108,10 @@ mod tests {
         };
         let mut tr: TestRead = Default::default();
         unsafe {
-            assert!(read_struct(&mut Cursor::new(source), &mut tr).is_err());
-            format!("{:?}", read_struct(&mut Cursor::new(source), &mut tr));
+            assert_eq!(
+                "Err(ReadStruct)", 
```

Comment:
Not related to this PR, but it looks like we are ignoring the error message returned by `read_exact` by doing `map_err(|_| Error::ReadStruct)?;`.

We should propagate the inner error by changing `ReadStruct` error definition:
```rust
use std::io;

...

#[derive(Debug)]
pub enum Error {
    ReadStruct(io::Error),
}
```

Then we should also implement `Display` for the error because Err(ReadStruct) is not very user friendly.

Do you want to also make these changes? If not I will open a different issue for this part.

Reply:
Of course, I would make the change!

Reply:
Agree, I've noticed this part, and I'd like to refactor the self-defined errors.

Reply:
@xiekeyang do you want to make the changes in this PR?

Reply:
Yes, I've amend and push it, PTAL after CI passed.